gh-release

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+#
+# This script reads the latest release on GitHub, checks out its
+# tag and runs a build command (defaults to `make archive`) with GOOS / GOARCH
+# combinations defined in the script. After that, it attaches the resulting
+# tarball to the release.
+
+set -eu -o pipefail
+
+TARGETS=("GOOS=linux  GOARCH=386" \
+         "GOOS=linux  GOARCH=arm" \
+         "GOOS=linux  GOARCH=amd64" \
+         "GOOS=darwin GOARCH=amd64")
+
+get_repo() {
+  local url=$(git config --get remote.origin.url)
+  if echo "$url" | grep -s '^https://' >/dev/null; then
+    echo "$url" | cut -d/ -f4,5
+  else
+    echo "$url" | cut -d: -f2 | cut -d. -f1
+  fi
+}
+
+fatal() {
+  echo "$@" >&2
+  exit 1
+}
+
+if [[ -z "${GH_TOKEN:-}" ]] && [[ ! -e "$HOME/.github-token" ]]; then
+  fatal "Either GH_TOKEN env var or ~/.github-token required"
+fi
+
+readonly TOKEN=${GH_TOKEN-$(cat ~/.github-token)}
+readonly CMD=${*-make archive}
+readonly REPO=$(get_repo)
+[[ -z "$REPO" ]] && fatal "Couldn't find remote url, no git repo?"
+
+readonly NAME=$(echo "$REPO" | cut -d/ -f2)
+readonly OLD_BRANCH=$(git branch | grep '^*'|cut -d' ' -f2)
```

Comment:
I think `git rev-parse --abbrev-ref HEAD` is equivalent to this, or does it behave subtly different in a wrong way?